packages/manager/apps/pci-workflow/src/api/hooks/order.ts

```diff
@@ -42,3 +50,56 @@ export const useInstanceSnapshotPricing = (
     data: snapshotPricing,
   };
 };
+
+export const findLocalZoneInstanceSnapshotPricing = (
+  projectId: string,
+  instanceRegion: string,
+  ovhSubsidiary: string,
+) => {
+  const queries = useQueries({
+    queries: [
+      { ...getCatalogQuery(ovhSubsidiary), enabled: Boolean(ovhSubsidiary) },
+      {
+        ...getProductAvailabilityQuery(projectId, ovhSubsidiary),
+        enabled: Boolean(projectId) && Boolean(ovhSubsidiary),
+      },
+    ],
+  });
+  const [catalog, productAvailability] = queries;
+  const snapshotPlanCode = productAvailability.data?.plans?.find(
+    ({ code, regions }) =>
+      code.startsWith('snapshot.consumption') &&
+      regions.find(({ name }) => name === instanceRegion),
+  )?.code;
+  const snapshotAddon =
+    catalog.data?.addons?.find(
+      (addon) => addon.planCode === snapshotPlanCode,
+    ) ||
+    catalog.data?.addons?.find(
+      (addon) => addon.planCode === 'snapshot.consumption.LZ',
+    );
```

Comment:
Could be simplified:
```suggestion
  const snapshotAddon = catalog.data?.addons?.find(
    (addon) =>
      addon.planCode === snapshotPlanCode ||
      addon.planCode === 'snapshot.consumption.LZ',
  );
```

Reply:
This is not the same algorithm
The first one acts as a fallback if the `snapshotPlanCode` it not found in the first place, witch is what we want to achieve. With your suggestion, we could find the `snapshot.consumption.LZ` before the `snapshotPlanCode` depending on the order of the addons and we could end up with an incorrect price

Reply:
Yes, it's another algorithm, thanks for the clarification :)